### ISSUE TITLE
IP range input support (ipv4, ipv6)

### ISSUE THREAD

**Issue by ehsandeep** [projectdiscovery/mapcidr#98]
Currently, IP or CIDR-based input is supported which can be extended to support **range-based** IP input which later can be utilized by other features like expansion/aggregation and other options.

Input: `192.168.0.1-192.168.0.255`

```console
echo 192.168.0.1-192.168.0.255 | mapcidr -a

192.168.0.1/32
192.168.0.2/31
192.168.0.4/30
192.168.0.8/29
192.168.0.16/28
192.168.0.32/27
192.168.0.64/26
192.168.0.128/25
```

Example is taken from https://www.ipaddressguide.com/cidr

<img width="416" alt="image" src="https://user-images.githubusercontent.com/8293321/187240448-20679444-3d48-425e-b4fc-dbf0ba038f03.png">

**Comment by jdk2588** [comment]
Something similar like this is already implemented in https://pkg.go.dev/inet.af/netaddr#IPSet 

It was proposed in net/ip.go here https://github.com/golang/go/issues/53370, but redirected to the above link. 

```
func main() {
	var b netaddr.IPSetBuilder

	b.AddRange(netaddr.IPRangeFrom(
		netaddr.MustParseIP("43.230.180.0"),
		netaddr.MustParseIP("43.230.182.255"),
	))
        
       	b.AddRange(netaddr.IPRangeFrom(
		netaddr.MustParseIP("2c0f:fec9::"),
		netaddr.MustParseIP("2c0f:fed7:ffff:ffff:ffff:ffff:ffff:ffff"),
	))

	s, _ := b.IPSet()

	fmt.Println("Ranges:")
	for _, r := range s.Ranges() {
		fmt.Printf("  %s - %s\n", r.From(), r.To())
	}

	fmt.Println("Prefixes:")
	for _, p := range s.Prefixes() {
		fmt.Printf("  %s\n", p)
	}

}

```

This produces output as 

```
Ranges:
  43.230.180.0 - 43.230.182.255
  2c0f:fec9:: - 2c0f:fed7:ffff:ffff:ffff:ffff:ffff:ffff
Prefixes:
  43.230.180.0/23
  43.230.182.0/24
  2c0f:fec9::/32
  2c0f:feca::/31
  2c0f:fecc::/30
  2c0f:fed0::/29
```

**Comment by jdk2588** [comment]
https://github.com/inetaf/netaddr is now deprecated. Instead should use https://github.com/go4org/netipx. 
Some of the benefits listed are here https://tailscale.com/blog/netaddr-new-ip-type-for-go/ 

TL; DR 


Feature | Go's net.IP
-- | --
Immutable | ❌, slice
Comparable | ❌, slice
Small | ❌, 28-56 bytes
Allocation free | ❌, slice's underlying array
Supports IPv4 & IPv6 | ✅
Can distinguish IPv4/IPv6 | ❌, #37921
Supports IPv6 zones | ❌, has separate net.IPAddr type
Opaque type | ❌, defined as []byte
Interops with standard library | ✅

**Comment by ShubhamRasal** [comment]
I found [rangeToCIDRrangeToCIDRs(firstIP, lastIP net.IP) []*net.IPNet](https://github.com/projectdiscovery/mapcidr/blob/d3d48b7c0f008b1cbc09bff781b51417e8e0ed4d/ip.go#L550)  which fulfills the requirements.
We just need to expose this function. 
I'll add the test cases to validate it.  Will that work? @ehsandeep
